rasa/nlu/classifiers/diet_classifier.py

```diff
@@ -706,6 +706,12 @@ def _create_model_data(
                 example for example in training_data if label_attribute in example.data
             ]
 
+        training_data = [
+            example
+            for example in training_data
+            if example.features_present(attribute=TEXT)
```

Comment:
yep good start 👍  -- we should also check that the features that are actually used by diet are checked for here (there should be a list of features in the config under something like `FEATURIZERS`) (and then we need to do the same in the `process` method)

Reply:
This part of the code is called during both training and prediction

Reply:
Finally realised that this must not be the same in training and prediction because otherwise we return less predictions than we requested. .... And I just realise this change is outdated so probably already fixed 👀